creator-node/src/utils.js

```diff
@@ -76,27 +78,43 @@ async function getIPFSPeerId (ipfs, config) {
 /** Cat single byte of file at given filepath. If ipfs.cat() call takes longer than the timeout time or
  * something goes wrong, an error will be thrown.
 */
-const ipfsSingleByteCat = (path, req, timeout = 1000) => new Promise(async (resolve, reject) => {
-  const start = Date.now()
-  let ipfs = req.app.get('ipfsLatestAPI')
+const ipfsSingleByteCat = (path, logContext, timeout = 1000) => {
+  const ipfsClient = require('ipfs-http-client-latest')
+  const ipfs = ipfsClient({
```

Comment:
the problem with that is the bull queue tries to serialize `req.app.get('ipfsLatestAPI')` into the redis cache before being able to process the task in the queue. it errors with `TypeError: Converting circular structure to JSON`. recreating the ipfs instance with the params was one of the ways to resolve this

Reply:
Yeah -- if the method is running from within the context of a bull queue operation, it will need to create its own IPFS client. However, this method is called outside of the bull queue context as well and we shouldn't recreate an ipfs client for that.

Reply:
Also, the queue first calls the async rehydrate method which is already instantiating its own IPFS client -- we shouldn't do that twice.

Reply:
ipfsSingleByteCat also gets called directly in /ipfs/:CID and /ipfs/:dirCID/:filename. perhaps we can make a global ipfsClient instead?

Reply:
discussed offline - will make ipfs global as to not need reinstantiation multiple times when method is called